### PR TITLE
Improve doc site left navbar structure

### PR DESCRIPTION
**Pull request by gregestren**
I used git poorly and accidentally broke https://github.com/bazelbuild/bazel/pull/10775.

This is the continuation of that.

Fixes https://github.com/bazelbuild/bazel/issues/10792

Preview: https://htmlpreview.github.io/?https://github.com/gregestren/bazel-docsy/blob/master/content/Bazel%20Overview%20-%20Bazel%202.1.0.htm

### CODE REVIEW

File: site/_layouts/documentation.html [hunk 1]
```diff
@@ -49,27 +49,32 @@ <h1>Documentation</h1>
...
+                <a href="/versions/{{ current_version }}/install.html">Overview</a>
+	      </li>
+
+
               <li>
                 <a class="sidebar-nav-heading" data-toggle="collapse"
                     href="#installing-menu" aria-expanded="false"
                     aria-controls="installing-menu">
-                  Installing Bazel<span class="caret"></span>
+                  Instructions<span class="caret"></span>
```
Comment (jin) [inline site/_layouts/documentation.html]: "Setting up"?

File: site/_layouts/documentation.html [hunk 2]
```diff
@@ -81,36 +86,36 @@ <h3>Installing and Using Bazel</h3>
...
                 <ul class="collapse sidebar-nav sidebar-submenu" id="using-menu">
-                  <li><a href="/versions/{{ current_version }}/build-ref.html">Bazel Concepts</a></li>
-                  <li><a href="/versions/{{ current_version }}/guide.html">User's Guide (Bazel commands)</a></li>
+                  <li><a href="/versions/{{ current_version }}/build-ref.html">Basic Concepts</a></li>
                   <li><a href="/versions/{{ current_version }}/external.html">External Dependencies</a></li>
                   <li><a href="/versions/{{ current_version }}/configurable-attributes.html">Configurable Attributes</a></li>
-                  <li><a href="/versions/{{ current_version }}/best-practices.html">Best Practices</a></li>
-                  <li><a href="/versions/{{ current_version }}/memory-saving-mode.html">Memory-saving Mode</a></li>
-                  <li><a href="/versions/{{ current_version }}/windows.html">Using Bazel on Windows</a></li>
+		  <li><a href="/versions/master/platforms-intro.html">Platforms</a></li>
```
Comment (jin) [inline site/_layouts/documentation.html]: indentation

File: site/_layouts/documentation.html [hunk 3]
```diff
@@ -268,7 +284,23 @@ <h3>Rules</h3>
...
+                  BUILD Files<span class="caret"></span>
+                </a>
+                <ul class="collapse sidebar-nav sidebar-submenu" id="build-files-menu">
+		  <li><a href="/versions/{{ current_version }}/be/functions.html">Functions</a></li>
+                  <li><a href="/versions/{{ current_version }}/be/common-definitions.html">Common Definitions</a></li>
+                  <li><a href="/versions/{{ current_version }}/be/make-variables.html">"Make" Variables</a></li>
+		  <li><a href="/versions/{{ current_version }}/skylark/build-style.html">BUILD Style Guide</a></li>
+                </ul>
+              </li>
+
```
Comment (jin) [inline site/_layouts/documentation.html]: indentation, and good idea for a dedicated BUILD files section :)

**Comment by gregestren** [comment]
Updated with feedback and a few more tweaks:

- Follow "**Sentence style**" capitalization guidelines: https://developers.google.com/style/capitalization#capitalization-in-titles-and-headings_1
- Try to keep titles concise
- But expand them out to make them reasonably precise (i.e. not vaguely generic)

There's a balance for that last point. I tried the principle of each section in isolation (i.e. an entry and its sub-entries) being reasonably self-contained and clear.

File: site/_layouts/documentation.html (hunk 1 above)
Comment (gregestren) [reply site/_layouts/documentation.html]: PTAL.

File: site/_layouts/documentation.html (hunk 2 above)
Comment (gregestren) [reply site/_layouts/documentation.html]: Done.

File: site/_layouts/documentation.html (hunk 3 above)
Comment (gregestren) [reply site/_layouts/documentation.html]: Done.

**Comment by gregestren** [comment]
Preview: https://htmlpreview.github.io/?https://github.com/gregestren/bazel-docsy/blob/master/content/Bazel%20Overview%20-%20Bazel%202.1.0.htm

File: site/_layouts/documentation.html [hunk 4]
```diff
@@ -125,58 +129,69 @@ <h3>Installing and Using Bazel</h3>
...
                 </a>
                 <ul class="collapse sidebar-nav sidebar-submenu" id="query-menu">
-                  <li><a href="/versions/{{ current_version }}/query-how-to.html">Bazel query</a></li>
-                  <li><a href="/versions/{{ current_version }}/cquery.html">Bazel cquery</a></li>
-                  <li><a href="/versions/{{ current_version }}/aquery.html">Bazel aquery</a></li>
-                  <li><a href="/versions/{{ current_version }}/query.html">Query Language</a></li>
+                  <li><a href="/versions/{{ current_version }}/query-how-to.html">bazel query</a></li>
+                  <li><a href="/versions/{{ current_version }}/cquery.html">bazel cquery</a></li>
+                  <li><a href="/versions/{{ current_version }}/aquery.html">bazel aquery</a></li>
+                  <li><a href="/versions/{{ current_version }}/query.html">Query language</a></li>
```
Comment (jin) [inline site/_layouts/documentation.html]: s/bazel query/The query command/ for all

**Pull request closed without merging**